ProjectHive/ProjectHive.Services.AuthAPI/Services/UserService.cs

```diff
@@ -0,0 +1,81 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using ProjectHive.Services.AuthAPI.Data;
+using ProjectHive.Services.AuthAPI.Data.Entities;
+using ProjectHive.Services.AuthAPI.Data.Repository.Interface;
+using ProjectHive.Services.AuthAPI.Dto;
+using ProjectHive.Services.Core.Business;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace ProjectHive.Services.AuthAPI.Services;
+
+public class UserService : Service<UserDto, User, ProjectHiveAuthDbContext>, IUserService
+{
+    private readonly IUserRepository userRepository;
+    private readonly IUnitOfWork unitOfWork;
```

Comment:
Preferred name is _userRepository for private fields or again - primary constructor